airflow-core/src/airflow/migrations/versions/0049_3_0_0_remove_pickled_data_from_xcom_table.py

```diff
@@ -77,9 +77,24 @@ def upgrade():
     condition = condition_templates.get(dialect)
     if not condition:
         raise RuntimeError(f"Unsupported dialect: {dialect}")
-
     # Key is a reserved keyword in MySQL, so we need to quote it
     quoted_key = conn.dialect.identifier_preparer.quote("key")
+    if dialect == "postgresql":
+        curr_timeout = (
+            int(
+                list(  # noqa: RUF015
+                    conn.execute(
+                        text("""select setting
+        from pg_settings
+        where name = 'statement_timeout'""")
+                    )
+                )[0][0]
+            )
+            / 1000
+        )
```

Comment:
This is a little unwieldy, can it be simplified?

```suggestion
        curr_timeout = int(
                conn.execute(
                        text("""select setting from pg_settings where name = 'statement_timeout'""")
                ).scalar_one()
        ) / 1000
```

Something like this maybe (untested)?

Reply:
i'll try it